### ISSUE TITLE
Forcing camp compile/install

### ISSUE THREAD

**Issue by tepperly** [LLNL/RAJA#845]
It would be nice to have a way to force RAJA to build/install camp through a cmake -D flag. I don't want to risk having find_package(camp QUIET) locate a system-installed or old install of camp. Thanks.

**Comment by tepperly** [comment]
I guess the same applies to cub and rocprim.

**Comment by corbett5** [comment]
I recently ran into an issue trying to build an application with Umpire 4.0.1 and RAJA 0.11.0 because they both had different versions of the camp submodule. I fixed this by updating RAJA to 0.12.1 however it would be nice to be able to pass `camp_DIR` to Umpire from the RAJA installation so that any problems would be caught when building Umpire and not the application itself.

**Comment by rhornung67** [comment]
@davidbeckingsale would you recommend building both RAJA and Umpire against the same external camp; i.e., neither uses camp submodule?

**Comment by davidbeckingsale** [comment]
That would be the easiest solution. There are too many combinations here for us to support robustly. I would also recommend being careful using a newer/older camp version though. We co-ordinated our RAJA, Umpire & CHAI releases to ensure they all worked together using the tagged camp release.

**Comment by corbett5** [comment]
That would definitely solve the problem, although I would prefer to just say for example the camp I will use will be the one built with RAJA then point all the other libraries at that installation since it means one fewer TPL that I need to maintain.

**Comment by davidbeckingsale** [comment]
> That would definitely solve the problem, although I would prefer to just say for example the camp I will use will be the one built with RAJA then point all the other libraries at that installation since it means one fewer TPL that I need to maintain.

Yup, but I wouldn't recommend that if you are using an old RAJA...

**Comment by corbett5** [comment]
Haha yeah for sure. But at that point you might run into problems when building your application if (as happened to me) Umpire's includes get routed to RAJA's camp. Having RAJA install a cmake package for camp would allow those errors to be worked out earlier when building Umpire.

**Comment by gzagaris** [comment]
Does `camp` include version information with its exported targets and/or header?

That may help in ensuring that the correct/same version of `camp` is used in CMake at the host application level and catch that error early. If nothing else, it can help with diagnosing these type of errors....

**Comment by trws** [comment]
It  currently does not, but it should.  I'll add versioning info into the installed target and a macro soon.

**Comment by rhornung67** [comment]
@trws can this issue be closed?

**Comment by trws** [comment]
I think it can?  A user can select a build of camp, or an external source dir, or I think even the built in one.  Does this remain an issue for you @tepperly?

**Comment by rhornung67** [comment]
Closing. Will open a new issue if issue resurfaces.
